test/testhelpers/loggerSpy.ts

```diff
@@ -0,0 +1,20 @@
+import { logger } from '../../src/utils/FSHLogger';
+
+class LoggerSpy {
+  private mockWriter = jest.spyOn(logger.transports[0], 'write');
+
+  getMessageAtIndex(index: number, reverse = false): string {
+    const i = reverse ? this.mockWriter.mock.calls.length - index - 1 : index;
```

Comment:
Good thinking to support a way to do this from the end of the message array.

In many JS functions that take an index (for example, [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice)), they use negative numbers to indicate offset from the end (e.g., reverse).  I'd suggest we follow suit, as it will be familiar to other JS developers.

So instead of:
```
expect(loggerSpy.getMessageAtIndex(1, true)).toMatch(/File: Dupe\.fsh.*Line: 7\D/s);
```
you have:
```
expect(loggerSpy.getMessageAtIndex(-1)).toMatch(/File: Dupe\.fsh.*Line: 7\D/s);
```

Reply:
What would I then do if I wanted the last element? For example, the case where right now we would do:

```
loggerSpy.getMessageAtIndex(0, true);
```

Reply:
Perhaps that could be in index of `-1` for the last element, and `-2` could be the second to last element?

Reply:
I decided to go with that approach. @cmoesel what do you think of the newest push?

Reply:
I'll look at the approach, but yeah -- `-1` should be the last element and `-2` is the second-to-last, etc.

Reply:
Yep -- looks good to me!